components/Button/Button.tsx

```diff
@@ -0,0 +1,44 @@
+import clsx from "clsx";
+import { on } from "events";
+
+interface ButtonProps {
+  bgColor: "primary" | "secondary" | "BG" | "error";
+  text: string;
+  width?: string;
+  py?: number;
+  clickFunction?: Function;
+}
+
+const Button = ({
+  bgColor,
+  text,
+  width = "full",
+  py = 0.5,
+  clickFunction,
+}: ButtonProps) => {
+  return (
+    <button
+      // we tried to write like below, but it caused an error with next-js style rendering
+      //   className={`bg-${bgColor} text-${bgColor}-text px-2 py-1 rounded-md`}
+      //   so we used a package clsx which generates strings from tournaries
+      // see more: https://github.com/lukeed/clsx/blob/master/readme.md
+      className={clsx(
+        bgColor === "primary"
+          ? "bg-primary text-primary-text"
+          : bgColor === "secondary"
+          ? "bg-secondary text-secondary-text"
+          : bgColor === "error"
+          ? "bg-error text-error-text"
+          : bgColor === "BG"
+          ? "bg-BG text-BG-text"
+          : "primary",
+        `px-8 py-${py} rounded-md w-${width}`
```

Comment:
interpolation in  tailwind classes doesnt work consistently, decide one some values you want to provide and use them in a ternary.

Here's the section in the docs that clarifies that:
https://tailwindcss.com/docs/content-configuration#dynamic-class-names

Reply:
interpolation seems to work if its wrapped in clsx(), so i refactored to take advantage of that. 
i also added a fixed height to the description box (i'm developing for iphone SE)